c10/core/SymIntArrayRef.h

```diff
@@ -13,6 +13,337 @@
 namespace c10 {
 using SymIntArrayRef = ArrayRef<SymInt>;
 
+template <>
+class ArrayRef<SymInt> final {
+  using T = SymInt;
+
+ public:
+  using iterator = const T*;
+  using const_iterator = const T*;
+  using size_type = size_t;
+  using value_type = T;
+
+  using reverse_iterator = std::reverse_iterator<iterator>;
+
+  enum Unchecked { UNCHECKED };
+
+ private:
+  /// The start of the array, in an external buffer.
+  const T* Data;
+
+  /// The number of elements.
+  size_type Length;
+
+  /// A bool saying if any element is symbolic
+  bool AnySymbolic;
+
+  void debugCheckNullptrInvariant() {
+    TORCH_INTERNAL_ASSERT_DEBUG_ONLY(
+        Data != nullptr || Length == 0,
+        "created ArrayRef with nullptr and non-zero length! c10::optional relies on this being illegal");
+  }
+
+  void debugCheckAnySymbolicInvariant() {
```

Comment:
nit: should this and the one above be marked inline? Or there is no way the compiler will miss these?

Reply:
Since they compile to empty bodies sans DEBUG I think it's safe to say the compiler will get it